heron/stmgr/src/cpp/manager/instance-server.cpp

```diff
@@ -371,16 +377,24 @@ void InstanceServer::SendToInstance2(sp_int32 _task_id,
 void InstanceServer::DrainTupleSet(sp_int32 _task_id,
                                 proto::system::HeronTupleSet2* _message) {
   TaskIdInstanceDataMap::iterator iter = instance_info_.find(_task_id);
-  if (iter == instance_info_.end() || iter->second->conn_ == NULL) {
-    LOG(ERROR) << "task_id " << _task_id << " has not yet connected to us. Dropping...";
-    if (_message->has_control()) {
+  if (iter == instance_info_.end() || iter->second->conn_ == NULL
+      || (droptuples_upon_backpressure_ && iter->second->conn_->hasCausedBackPressure())) {
+    LOG(ERROR) << "task_id " << _task_id
+               << " has not yet connected to us or is not keeping up, Dropping...";
```

Comment:
same, suggest to add "and droptuples_upon_backpressure is set to true" in log.

Reply:
Added